hutch_python/_startup_script.py

```diff
@@ -0,0 +1,40 @@
+"""
```

Comment:
I'm glad this is pulled out into a lintable file, but after ruminating a bit I don't think that any of these need to be deferred until the exec_lines hook. Can't all of these be ipython extensions instead?

What are the advantages of doing anything library-side in the exec_lines?

I think I have some built-in skepticism whenever I see strings being exec'd

Reply:
Yea I'm not a fan of exec-ing stuff either, and I seem to be doing it a lot..

The ipython extensions we set up in the traitlets config cannot access the namespace items (devices, RE, etc).  That's really the primary reason for this functionality.  Things like the elog callback require the elog instance as well as the ipython input record for initialization.  

I'll admit I overlooked the ipython extensions earlier, and 2/3 of the configurations I singled out can be moved there

Reply:
Gotcha, there's no way at all to pass resources into IPython extensions, except gross globals, maybe? That's a bummer.